### PR TITLE
refine d.ts

### PR DESCRIPTION
**Pull request by xduooo**
Re: cocos-creator/2d-tasks#

Changes:
 * Some data types or return values are normalized to prevent error prompts.

### CODE REVIEW

**Comment by SantyWang** [comment]
Cool, Thank you very much!

**Comment by xduooo** [comment]
There are still mistakes I haven't modified. Please see the error list.

base: branch v2.4.0  [creator_engine_dts_errs.zip](https://github.com/cocos-creator/engine/files/4468145/creator_engine_dts_errs.zip)

File: cocos2d/core/value-types/color.ts [hunk 1]
```diff
@@ -185,7 +185,7 @@ export default class Color extends ValueType {
      * Set the components of a color to the given values.
      * @method set
      * @typescript
-     * set (out: Color, r = 255, g = 255, b = 255, a = 255): Color
+     * set (out: Color, r?: number, g?: number, b?: number, a?: number): Color
```
Comment (2youyou2) [inline cocos2d/core/value-types/color.ts:188]: Why to remove the default param value

Comment (xduooo) [reply cocos2d/core/value-types/color.ts:188]: When we compile typescript with `tsc`command, we will get this error :

> error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

Comment (xduooo) [reply cocos2d/core/value-types/color.ts:188]: Microsoft is also discussing a better way to express default values ( https://github.com/Microsoft/tsdoc/issues/151 ),  but now it's not the right way:
`set (out: Color, r = 255, g = 255, b = 255, a = 255): Color
`

Comment (2youyou2) [reply cocos2d/core/value-types/color.ts:188]: I see, thanks.

**Review by 2youyou2 — APPROVED** [review]

File: cocos2d/actions/tween.js [hunk 2]
```diff
@@ -215,9 +215,9 @@ Tween.stopAllByTarget = function (target) {
  * Insert an action or tween to this sequence
  * !#zh
  * 插入一个 action 或者 tween 到队列中
- * @method then 
- * @param {Action|Tween} other
- * @return {Tween}
+ * @method then
+ * @param {Action|Tween<T>} other
+ * @return {Tween<T>}
```
Comment (SantyWang) [inline cocos2d/actions/tween.js:220]: I don't think api-document-generator can recognize genericity in `param` and `return`. It's better to add a decorator 'typescript' likes below
```
@typescript then (othen: Action|Tween<T>): Tween<T>
```

Comment (xduooo) [reply cocos2d/actions/tween.js:220]: I have tested it with the following tools, it works:
https://github.com/cocos-creator/creator-api-docs and
https://github.com/cocos-creator/firedoc

It is an internal workflow, so you can decide which one is better.

**Comment by zhefengzhang** [comment]
I dealt with this issue on the basis of this PR and submitted a new PR：
image

**Comment by SantyWang** [comment]
Thanks for your contrubution

**Pull request merged**